src/EditorFeatures/Core.Wpf/StringCopyPaste/WpfStringCopyPasteService.cs

```diff
@@ -0,0 +1,160 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Composition;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Windows;
+using Microsoft.CodeAnalysis.ErrorReporting;
+using Microsoft.CodeAnalysis.Host.Mef;
+
+namespace Microsoft.CodeAnalysis.Editor.StringCopyPaste
+{
+    [ExportWorkspaceService(typeof(IStringCopyPasteService), ServiceLayer.Host), Shared]
+    internal class WpfStringCopyPasteService : IStringCopyPasteService
```

Comment:
@gundermanc This type is pretty scary/sketchy.  I don't love intercepting 'copy' in this fashion to add our own flavors to it.  If editor/platform could expose a way to add info when it was handling copy itself, that would be helpful and less scary.

Reply:
Please let me know if you have any specific suggestions. Otherwise we could file an experience to design some API to make this simpler.

Reply:
the general need we have is to add our own key/value to teh clipboard when a copy/paste happens.  right now we're doing htat by allowing *you* to copy/paste.  We then copy all of the data off hte clipboard and create our own dataobject to put back on, with teh copy of everything *and* our new KVP.